account_payment_order/models/account_move.py

```diff
@@ -23,9 +23,9 @@ class AccountMove(models.Model):
     # payment mode or company level
     reference_type = fields.Selection(
         selection=[("none", "Free Reference"), ("structured", "Structured Reference")],
-        readonly=True,
```

Comment:
Consider adding `reference_type` to the `_compute_payment_reference` dependencies. We didn't need it before because the field was read-only after the invoice was posted. Now, since it can change for posted invoices, we need to trigger the payment reference recompute.